velox/functions/sparksql/specialforms/GetStructField.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/expression/FunctionCallToSpecialForm.h"
+#include "velox/expression/SpecialForm.h"
+
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::functions::sparksql {
+
+class GetStructFieldExpr : public SpecialForm {
```

Comment:
Maybe you need to extend class `FunctionCallToSpecialForm` rather than `SpecialForm`, so as the implementation

Reply:
I think there are different, one is for function register, the other is for function iteself

Reply:
I'm wondering if we can implement the get_struct_field as a vector function then register it as a special form. What's the specific consideration here to implement it by extending `SpecialForm`?